### PR TITLE
Pass context to transform

### PR DESCRIPTION
**Pull request by flow-jacques**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ Read more:
 
 ## License
 
-Copyright © 2019–2022 Mike Fikes
+Copyright © 2019–2022 Mike Fikes.
```
Comment (flow-danny) [inline README.md]:
```suggestion
Copyright © 2019–2022 Mike Fikes
```

File: src/cljs_bean/core.cljs [hunk 2]
```diff
@@ -15,15 +18,15 @@
...
+                          (transform x ctx))]
     transformed
     (cond
       (primitive? x) x
-      (object? x) (Bean. nil x prop->key key->prop transform true nil nil nil)
-      (array? x) (ArrayVector. nil prop->key key->prop transform x nil)
-      :else x)))
+      (object? x)    (Bean. nil x prop->key key->prop transform true nil nil nil)
+      (array? x)     (ArrayVector. nil prop->key key->prop transform x nil)
+      :else          x)))
```
Comment (flow-danny) [inline src/cljs_bean/core.cljs]: Nothing was changed?
```suggestion
      (object? x) (Bean. nil x prop->key key->prop transform true nil nil nil)
      (array? x) (ArrayVector. nil prop->key key->prop transform x nil)
      :else x)))
```

File: src/cljs_bean/core.cljs [hunk 3]
```diff
@@ -7,6 +7,9 @@
 (declare ArrayVector)
 (declare ->clj)
 
+(defrecord TransformContext
+    [prop key nth])
+
 (def ^:private lookup-sentinel #js {})
 
 (defn- primitive? [x]
```
Comment (flow-danny) [inline src/cljs_bean/core.cljs]: Maybe unprivate this
```suggestion
(defn primitive? [x]
```

File: src/cljs_bean/core.cljs [hunk 4]
```diff
@@ -292,17 +308,19 @@
     (if (snapshot? k v prop->key recursive?)
       (-assoc (with-meta (snapshot obj prop->key key->prop transform recursive?) meta) k v)
       (Bean. meta
-        (doto (gobj/clone obj) (unchecked-set (key->prop k) (cond-> v recursive? unwrap)))
+             (doto (gobj/clone obj) (unchecked-set (key->prop k) (cond-> v recursive? unwrap)))
```
Comment (flow-danny) [inline src/cljs_bean/core.cljs]:
```suggestion
        (doto (gobj/clone obj) (unchecked-set (key->prop k) (cond-> v recursive? unwrap)))
```

File: src/cljs_bean/core.cljs [hunk 5]
```diff
@@ -401,8 +429,8 @@
           (== n (alength arr)) (-conj! tcoll val)
           :else
           (throw
-            (js/Error.
-              (str "Index " n " out of bounds for TransientArrayVector of length" (alength arr))))))
+           (js/Error.
+            (str "Index " n " out of bounds for TransientArrayVector of length" (alength arr))))))
```
Comment (flow-danny) [inline src/cljs_bean/core.cljs]:
```suggestion
            (js/Error.
              (str "Index " n " out of bounds for TransientArrayVector of length" (alength arr))))))
```

File: src/cljs_bean/core.cljs [hunk 6]
```diff
@@ -568,12 +601,12 @@
...
-        :else
-        (let [new-arr (aclone arr)]
-          (ArrayVector. meta prop->key key->prop transform
-            (.slice new-arr 0 (dec (alength new-arr))) nil))))
+      (zero? (alength arr)) (throw (js/Error. "Can't pop empty vector"))
+      (== 1 (alength arr))  (-empty coll)
+      :else
+      (let [new-arr (aclone arr)]
+        (ArrayVector. meta prop->key key->prop transform
+                      (.slice new-arr 0 (dec (alength new-arr))) nil))))
```
Comment (flow-danny) [inline src/cljs_bean/core.cljs]:
```suggestion
        (zero? (alength arr)) (throw (js/Error. "Can't pop empty vector"))
        (== 1 (alength arr))  (-empty coll)
        :else
        (let [new-arr (aclone arr)]
          (ArrayVector. meta prop->key key->prop transform
            (.slice new-arr 0 (dec (alength new-arr))) nil))))
```

File: src/cljs_bean/core.cljs [hunk 7]
```diff
@@ -638,7 +674,7 @@
           (aset new-arr n (unwrap val))
           (ArrayVector. meta prop->key key->prop transform new-arr nil)))
       (== n (alength arr)) (-conj coll val)
-      :else (throw (js/Error. (str "Index " n " out of bounds  [0," (alength arr) "]")))))
+      :else                (throw (js/Error. (str "Index " n " out of bounds  [0," (alength arr) "]")))))
```
Comment (flow-danny) [inline src/cljs_bean/core.cljs]:
```suggestion
      :else (throw (js/Error. (str "Index " n " out of bounds  [0," (alength arr) "]")))))
```

**Review by flow-danny — CHANGES_REQUESTED** [review]

File: src/cljs_bean/core.cljs [hunk 8]
```diff
@@ -569,7 +602,7 @@
   (-pop [coll]
     (cond
         (zero? (alength arr)) (throw (js/Error. "Can't pop empty vector"))
-        (== 1 (alength arr)) (-empty coll)
+        (== 1 (alength arr))  (-empty coll)
```
Comment (flow-danny) [inline src/cljs_bean/core.cljs]:
```suggestion
        (== 1 (alength arr)) (-empty coll)
```

**Review by flow-danny — COMMENTED** [review]
Missed 1 whitespace non-change

File: src/cljs_bean/core.cljs (hunk 3 above)
Comment (flow-jacques) [reply src/cljs_bean/core.cljs]: Done

**Review by flow-danny — APPROVED** [review]

**Pull request merged**